www-client/seamonkey/seamonkey-2.53.20.ebuild

```diff
@@ -0,0 +1,553 @@
+	if tc-ld-is-gold ; then
+		mozconfig_annotate 'tc-ld-is-gold=true' --enable-gold
+	else
+		mozconfig_annotate 'tc-ld-is-gold=false' --disable-gold
+	fi
+
+	# Debug is broken, disable debug symbols
+	mozconfig_annotate 'disabled by Gentoo' --disable-debug-symbols
+
+	# wifi pulls in dbus so manage both here
+	mozconfig_use_enable wifi necko-wifi
+	if use kernel_linux && use wifi && ! use dbus ; then
+		echo "Enabling dbus support due to wifi request"
+		mozconfig_annotate 'dbus required by necko-wifi on linux' --enable-dbus
+	else
+		mozconfig_use_enable dbus
+		mozconfig_annotate 'disabled' --disable-necko-wifi
+	fi
+
+	# These are enabled by default in all mozilla applications
+	mozconfig_annotate '' --with-system-nspr
+	mozconfig_annotate '' --with-system-nss --with-nss-prefix="${ESYSROOT}"/usr
+	mozconfig_annotate '' --x-includes="${ESYSROOT}"/usr/include --x-libraries="${ESYSROOT}"/usr/$(get_libdir)
+	if use system-libevent ; then
+		mozconfig_annotate '' --with-system-libevent="${ESYSROOT}"/usr
+	fi
+	mozconfig_annotate '' --prefix="${EPREFIX}"/usr
+	mozconfig_annotate '' --libdir="${EPREFIX}"/usr/$(get_libdir)
+	mozconfig_annotate '' --disable-crashreporter
+	mozconfig_annotate '' --enable-system-ffi
+	mozconfig_annotate '' --with-intl-api
+
+	# default toolkit is cairo-gtk3, optional use flags can change this
+	mozconfig_annotate '' --enable-default-toolkit=cairo-gtk3
+
+	# Instead of the standard --build= and --host=, mozilla uses --host instead
+	# of --build, and --target intstead of --host.
+	# Note, mozilla also has --build but it does not do what you think it does.
+	# Set both --target and --host as mozilla uses python to guess values otherwise
+	mozconfig_annotate '' --target="${CHOST}"
+	mozconfig_annotate '' --host="${CBUILD:-${CHOST}}"
+
+	mozconfig_use_enable pulseaudio
+	# force the deprecated alsa sound code if pulseaudio is disabled
+	if use kernel_linux && ! use pulseaudio ; then
+		mozconfig_annotate '-pulseaudio' --enable-alsa
+	fi
+
+	mozconfig_use_enable system-sqlite
+	mozconfig_use_with system-jpeg
+	mozconfig_use_with system-icu
+	mozconfig_use_with system-libvpx
+	mozconfig_use_with system-png
+	mozconfig_use_with system-harfbuzz
+	mozconfig_use_with system-harfbuzz system-graphite2
+	mozconfig_use_with system-av1
+
+	# Modifications to better support ARM, bug 553364
+	if use cpu_flags_arm_neon ; then
+		mozconfig_annotate '' --with-fpu=neon
+		mozconfig_annotate '' --with-thumb=yes
+		mozconfig_annotate '' --with-thumb-interwork=no
+	fi
+	if [[ ${CHOST} == armv* ]] ; then
+		mozconfig_annotate '' --with-float-abi=hard
+		if ! use system-libvpx ; then
+			sed -i -e "s|softfp|hard|" media/libvpx/moz.build || die
+		fi
+	fi
+
+	if use lto ; then
+		# Linking only works when using ld.gold when LTO is enabled
+		mozconfig_annotate "forcing ld=gold due to USE=lto" --enable-linker=gold
+		# ThinLTO is currently broken, see bmo#1644409
+		mozconfig_annotate '+lto' --enable-lto=full
```

Comment:
Also we don't enable `lto` via use flags anymore, but via `tc-is-lto` function. Check firefox for example. Does `--enable-lto=full` work already without gold? I'm getting mixed feedback from the bug, someone even has a patch linked there.

Reply:
Replying here for both comments.
I don't know the state of lto/gold, nor use it myself. I also lack the knowledge about it, so this is a part I never touched. I'll have a look into this and check the firefox ebuild.

Reply:
I had contact with Seamonkey upstream, they suggested to drop support for lto. They don't build/test for it themselves and are not sure if it even works (as it should).

I'm considering to just do that. If people want to have lto support, they can always open a bug on bgo (or with upstream) and I'm willing to help them with that.

@juippis would that be acceptable?

Reply:
I would like you to at least write the logic using `tc-is-lto` that then enables the configure `mozconfig_annotate '+lto' --enable-lto=full` - like is done in firefox. I can test lto with seamonkey and if it doesn't work, let's just `filter-lto` and not allow it to be enabled.

If there's a chance it'll work, at least it's correctly written in the ebuild.

Reply:
Ok, than let's try to keep it in. Working on that now.